src/net/lwipopts.h

```diff
@@ -77,6 +77,7 @@ typedef unsigned long size_t;
 #define LWIP_DHCP_BOOTP_FILE 1
 #define LWIP_DNS 1
 #define LWIP_INCLUDED_POLARSSL_MD5  1
+#define LWIP_NETIF_STATUS_CALLBACK 1
```

Comment:
Instead of enabling an additional lwip option, we could just use the existing lwip_ext_callback function, i.e. add the ip4_complete logic in the code that handles the (reason & LWIP_NSC_IPV4_ADDRESS_CHANGED) condition